cmd/api-response.go

```diff
@@ -275,9 +275,26 @@ func getLocation(r *http.Request) string {
 	return path.Clean(r.URL.Path) // Clean any trailing slashes.
 }
 
-// getObjectLocation gets the relative URL for an object
-func getObjectLocation(bucketName string, key string) string {
-	return "/" + bucketName + "/" + key
+// returns "https" if the tls boolean is true, "http" otherwise.
+func getURLScheme(tls bool) string {
+	if tls {
+		return httpsScheme
+	}
+	return httpScheme
+}
+
+// getObjectLocation gets the fully URI of an object if
+// server is bound to a specific IP or hostname. If not
+// the URI is relative.
+func getObjectLocation(host, bucket, object string) string {
```

Comment:
We would need to use something like below
```go
u := URL{
Scheme: getURLScheme(),
Host: ...,
Path: ...,
}

return u.String()
```

Reply:
Not as default .. see the reasoning why it's based on `--address` alone. Otherwise we need it as relative..

Reply:
@balamurugana can you take look again?